core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -539,21 +539,53 @@ private void validateSenderAndReceiver()
       throw new BadRequestException("receiver_id cannot be empty.");
     }
 
-    // TODO: Populate customer type with the first receiver type; this is a temporary fix for cases
-    // where customerType is required
     Sep12Operation sep12Operation = Context.get().getAsset().getSep31().getSep12();
-    String receiverType = null;
+    Sep12GetCustomerResponse receiver;
+
+    // Fix for https://github.com/stellar/java-stellar-anchor-sdk/issues/717 (ANCHOR-149)
+    // TODO: check if there is a better way to do it, i.e. request a type first for a customer OR
+    // stop iteration after first hit.
```

Comment:
Is there a reason we're not checking the customer status after each request and stopping if we get ACCEPTED?

Reply:
I don't think I'm following the question, can you elaborate please?

Reply:
Right now the implementation makes all the requests, then checks all the statuses. Why don't we make a request & check the status before we make the next request?

Reply:
Do you suggest to request and check them one by one until ACCEPTED found?
We can do that. But if there is a misconfiguration  `Sep31AmbiguousCustomerInfoException` won't be thrown.

Reply:
Oh you're right. Ok, lets keep it as-is.